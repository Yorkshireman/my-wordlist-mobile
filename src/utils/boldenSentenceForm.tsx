import { Text } from 'react-native-paper';

export const boldenSentenceForm = ({
  content,
  form
}: {
  content: string;
  form?: string | null;
}) => {
  if (!form) return <Text>{content}</Text>;

  // Split the sentence into parts, where each part is either the form or not
  // g means it will not stop after the first match
  // i means it will be case-insensitive
  const parts = content.split(new RegExp(`(\\b${form}\\b)`, 'gi'));

  return (
    <Text>
      {parts.map((part, index) =>
        part.toLowerCase() === form.toLowerCase() ? (
          <Text key={index} style={{ fontWeight: 'bold' }}>
            {part}
          </Text>
        ) : (
          part
        )
      )}
    </Text>
  );
};
