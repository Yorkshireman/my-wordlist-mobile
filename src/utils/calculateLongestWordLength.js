export const calculateLongestWordLength = wordlistEntries => {
  if (!wordlistEntries) {
    return console.error('null/undefined wordlistEntries param passed to calculateLongestWordLength()');
  }

  return wordlistEntries.reduce((acc, entry) => {
    const wordLength = entry.word.text.length;
    return wordLength > acc ? wordLength : acc;
  }, 0);
};
