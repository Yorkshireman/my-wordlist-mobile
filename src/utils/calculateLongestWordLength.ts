export const calculateLongestWordLength = (
  wordlistEntries: { word: { text: string } }[]
): number => {
  return wordlistEntries.reduce((acc, entry) => {
    const wordLength = entry.word.text.length;
    return wordLength > acc ? wordLength : acc;
  }, 0);
};
