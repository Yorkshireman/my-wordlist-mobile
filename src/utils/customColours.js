export const customColours = {
  backdrop: 'rgba(55, 46, 52, 0.4)',
  background: 'rgb(255, 251, 255)',
  elevation: {
    level0: 'transparent',
    level1: 'rgb(250, 241, 250)',
    level2: 'rgb(247, 234, 247)',
    level3: 'rgb(244, 228, 244)',
    level4: 'rgb(243, 226, 243)',
    level5: 'rgb(241, 222, 241)'
  },
  error: 'rgb(186, 26, 26)',
  errorContainer: 'rgb(255, 218, 214)',
  inverseOnSurface: 'rgb(248, 238, 242)',
  inversePrimary: 'rgb(255, 170, 243)',
  inverseSurface: 'rgb(52, 47, 50)',
  onBackground: 'rgb(30, 26, 29)',
  onError: 'rgb(255, 255, 255)',
  onErrorContainer: 'rgb(65, 0, 2)',
  onPrimary: 'rgb(255, 255, 255)',
  onPrimaryContainer: 'rgb(56, 0, 56)',
  onSecondary: 'rgb(255, 255, 255)',
  onSecondaryContainer: 'rgb(39, 22, 36)',
  onSurface: 'rgb(30, 26, 29)',
  onSurfaceDisabled: 'rgba(30, 26, 29, 0.38)',
  onSurfaceVariant: 'rgb(78, 68, 75)',
  onTertiary: 'rgb(255, 255, 255)',
  onTertiaryContainer: 'rgb(50, 18, 8)',
  outline: 'rgb(128, 116, 124)',
  outlineVariant: 'rgb(209, 194, 203)',
  primary: 'rgb(132, 45, 180)',
  primaryContainer: 'rgb(255, 215, 245)',
  scrim: 'rgb(0, 0, 0)',
  secondary: 'rgb(109, 88, 105)',
  secondaryContainer: 'rgb(247, 218, 239)',
  shadow: 'rgb(0, 0, 0)',
  surface: 'rgb(255, 251, 255)',
  surfaceDisabled: 'rgba(30, 26, 29, 0.12)',
  surfaceVariant: 'rgb(238, 222, 231)',
  tertiary: 'rgb(130, 83, 69)',
  tertiaryContainer: 'rgb(255, 219, 209)'
};
