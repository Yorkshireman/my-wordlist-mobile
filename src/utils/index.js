export { calculateLongestWordLength } from './calculateLongestWordLength';
export { customColours } from './customColours';
export { isValidWordlistEntry } from './isValidWordlistEntry';
export { parseCategories } from './parseCategories';
export { storeAuthToken } from './storeAuthToken.js';
