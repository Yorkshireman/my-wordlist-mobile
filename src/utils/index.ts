export { boldenSentenceForm } from './boldenSentenceForm';
export { calculateLongestWordLength } from './calculateLongestWordLength';
export { customColours } from './customColours';
export { displayLanguage } from './displayLanguage';
export { isInvalidEmail } from './isInvalidEmail';
export { parseCategories } from './parseCategories';
export { parseUniqueCategories } from './parseUniqueCategories';
export { sanitiseText } from './sanitiseText';
export { signIn } from './signIn';
export { signUp } from './signUp';
export { storeAuthToken } from './storeAuthToken';
