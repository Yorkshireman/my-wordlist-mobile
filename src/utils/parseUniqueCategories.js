export const parseUniqueCategories = entries => {
  if (!entries) return null;

  const categories = entries.map(entry =>
    entry.categories.map(({ id, name }) => ({ id, name }))
  ).flat();

  return categories.reduce((acc, category) => {
    if (!acc.some(({ id }) => id === category.id)) {
      acc.push(category);
    }
    return acc;
  }, []).sort((a, b) => a.name > b.name ? 1 : -1);
};
