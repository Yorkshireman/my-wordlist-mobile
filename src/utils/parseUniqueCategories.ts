import { WordlistEntry } from '../__generated__/graphql';

export const parseUniqueCategories = (entries: WordlistEntry[]) => {
  if (!entries) return null;

  const categories = entries
    .map(entry => entry.categories.map(({ id, name }) => ({ id, name })))
    .flat();

  return categories
    .reduce((acc: { id: string; name: string }[], category) => {
      if (!acc.some(({ id }) => id === category.id)) {
        acc.push(category);
      }
      return acc;
    }, [])
    .sort((a, b) => (a.name > b.name ? 1 : -1));
};
