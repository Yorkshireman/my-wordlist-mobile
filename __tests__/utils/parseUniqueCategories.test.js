import { parseUniqueCategoriesMock as entries } from './parseUniqueCategoriesMock';
import { parseUniqueCategories } from '../../src/utils/parseUniqueCategories';

describe('parseUniqueCategories()', () => {
  test('leaves only unique categories with ids, in alphabetical order', () => {
    const result = parseUniqueCategories(entries);
    expect(result).toMatchSnapshot();
  });

  test('returns null when passed undefined', () => {
    expect(parseUniqueCategories()).toBeNull();
  });

  test('returns null when passed an empty array', () => {
    expect(parseUniqueCategories()).toBeNull();
  });
});
