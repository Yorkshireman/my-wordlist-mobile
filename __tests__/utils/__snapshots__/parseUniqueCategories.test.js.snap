// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseUniqueCategories() leaves only unique categories with ids, in alphabetical order 1`] = `
[
  {
    "id": "76b204c3-1194-454e-a4c6-3d1e281a95fe",
    "name": "adjective",
  },
  {
    "id": "82d5375f-9a34-4c2d-b9c2-8b24b88b066e",
    "name": "adverb",
  },
  {
    "id": "5d0c09ad-0f79-47b7-b2f3-6567ef5e6ad8",
    "name": "food",
  },
  {
    "id": "b67f419c-2556-450c-9a24-752ec32cf9c3",
    "name": "home",
  },
  {
    "id": "f6c0fc6f-3b01-4059-9588-ccb2bda39d23",
    "name": "household",
  },
  {
    "id": "fdedf5fe-745c-4a9e-81d9-31d2817276d4",
    "name": "nature",
  },
  {
    "id": "905651d6-2d66-44c3-9e89-7ef076afb6b5",
    "name": "noun",
  },
  {
    "id": "cc567ecc-4588-4584-871a-ee4a2fef6b7a",
    "name": "plural",
  },
  {
    "id": "44458ace-7679-49e0-b99a-a8738b1c248d",
    "name": "transport",
  },
  {
    "id": "0fd9e383-ca71-44bd-a73e-394c73c15a2e",
    "name": "verb",
  },
  {
    "id": "5097e71c-ba7f-4fcd-9122-91dadad10c88",
    "name": "weather",
  },
]
`;
